### PR TITLE
Add capability to switch b/w XMPP and FCM.

### PR DESCRIPTION
**Pull request by pendyala82**
The current CL enables connector to be able to switch printer notification channel b/w XMPP and FCM

Fcm channel is enabled if fcm flag is passed during connector start up , else connector falls back to XMPP channel.

Gcp will migrate printers b/w FCM and XMPP accordingly.

### CODE REVIEW

File: lib/printer.go [hunk 1]
```diff
@@ -22,25 +22,26 @@ type DuplexVendorMap map[cdd.DuplexType]string
...
+	Manufacturer        string                         // CUPS: PPD;                         GCP: manufacturer field
+	Model               string                         // CUPS: PPD;                         GCP: model field
+	GCPVersion          string                         //                                    GCP: gcpVersion field
+	SetupURL            string                         //                                    GCP: setup_url field
+	SupportURL          string                         //                                    GCP: support_url field
+	UpdateURL           string                         //                                    GCP: update_url field
+	ConnectorVersion    string                         //                                    GCP: firmware field
+	State               *cdd.PrinterStateSection       // CUPS: various;                     GCP: semantic_state field
+	Description         *cdd.PrinterDescriptionSection // CUPS: translated PPD;              GCP: capabilities field
+	CapsHash            string                         // CUPS: hash of PPD;                 GCP: capsHash field
```
Comment (adlr) [inline lib/printer.go]: looks like you added a bunch of whitespace before "GCP:" in this line. Mistake?

File: lib/printer.go [hunk 2]
```diff
@@ -219,11 +221,16 @@ func diffPrinter(pn, pg *Printer) PrinterDiff {
...
+	if pg.NotificationChannel != pn.NotificationChannel {
+		d.NotificationChannelChanged = true
+	}
+
 	if d.DefaultDisplayNameChanged || d.ManufacturerChanged || d.ModelChanged ||
-		d.GCPVersionChanged || d.SetupURLChanged || d.SupportURLChanged ||
-		d.UpdateURLChanged || d.ConnectorVersionChanged || d.StateChanged ||
-		d.DescriptionChanged || d.CapsHashChanged || d.TagsChanged ||
-		d.DuplexMapChanged || d.QuotaEnabledChanged || d.DailyQuotaChanged {
+			d.GCPVersionChanged || d.SetupURLChanged || d.SupportURLChanged ||
```
Comment (adlr) [inline lib/printer.go]: looks like some unnecessary whitespace changes in these lines?

Comment (pendyala82) [reply lib/printer.go]: they are auto formatted.

File: lib/printer.go (hunk 1 above)
Comment (pendyala82) [reply lib/printer.go]: they are auto formatted.
